packages/js/components/src/woo-product-tab-item/woo-product-tab-item.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+import React, { ReactElement, ReactNode } from 'react';
+import { Slot, Fill, TabPanel } from '@wordpress/components';
+import { createElement, Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { createOrderedChildren } from '../utils';
+
+type WooProductTabItemProps = {
+	id: string;
+	pluginId: string;
+	template?: string;
+	order?: number;
+	tabProps:
+		| TabPanel.Tab
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		| ( ( fillProps: Record< string, any > | undefined ) => TabPanel.Tab );
+	templates?: Array< { name: string; order?: number } >;
+};
+
+type WooProductFieldSlotProps = {
+	template: string;
+	children: (
+		tabs: TabPanel.Tab[],
+		tabChildren: Record< string, ReactNode >
+	) => ReactElement | null;
+};
+
+export const WooProductTabItem: React.FC< WooProductTabItemProps > & {
+	Slot: React.VFC<
+		Omit< Slot.Props, 'children' > & WooProductFieldSlotProps
+	>;
+} = ( { children, order, template, tabProps, templates } ) => {
+	if ( ! template && ! templates ) {
+		// eslint-disable-next-line no-console
+		console.warn(
+			'WooProductTabItem fill is missing template or templates property.'
+		);
+		return null;
+	}
+	templates = templates || [ { name: template as string, order } ];
```

Comment:
I like the structure here, although I wonder if it just makes it more confusing to support both the `template/order` and `templates` props APIs. I might be in favor of switching to just the `templates` version, and then do the same with the other slot fills. 

That said, this is totally optional.

Reply:
Just `templates` would be fine as well, I kinda figured the primary use case would still be a single location, which is why I kept the `template` & `order` props, but adding a single item to the `templates` array isn't to much of a difference.